packages/core/src/modules/oob/repository/OutOfBandRecord.ts

```diff
@@ -0,0 +1,81 @@
+import type { TagsBase } from '../../../storage/BaseRecord'
+import type { DidCommService } from '../../dids'
+import type { OutOfBandRole } from '../domain/OutOfBandRole'
+import type { OutOfBandState } from '../domain/OutOfBandState'
+
+import { Type } from 'class-transformer'
+
+import { AriesFrameworkError } from '../../../error'
+import { BaseRecord } from '../../../storage/BaseRecord'
+import { uuid } from '../../../utils/uuid'
+import { OutOfBandMessage } from '../messages'
+
+export interface OutOfBandRecordProps {
+  id?: string
+  createdAt?: Date
+  updatedAt?: Date
+  tags?: TagsBase
+  outOfBandMessage: OutOfBandMessage
+  role: OutOfBandRole
+  state: OutOfBandState
+  autoAcceptConnection: boolean
+  reusable?: boolean
+}
+
+export class OutOfBandRecord extends BaseRecord<TagsBase> {
+  @Type(() => OutOfBandMessage)
```

Comment:
I see you're using `OutOfBandMessage` instead of `OutOfBandInvitationMessage`. Is there a specific reason in deviating from the RFC?

Reply:
Fair point. No particular reason for that. Maybe, we can name it just `InvitationMessage` because it's already in oob module and also `HandshakeReuseMessage` is not `OutOfBandHandshakeReuseMessage`.